compiler/rustc_middle/src/query/mod.rs

```diff
@@ -433,12 +433,31 @@ rustc_queries! {
         /// full predicates are available (note that supertraits have
         /// additional acyclicity requirements).
         query super_predicates_of(key: DefId) -> ty::GenericPredicates<'tcx> {
-            desc { |tcx| "computing the supertraits of `{}`", tcx.def_path_str(key) }
+            desc { |tcx| "computing the super predicates of `{}`", tcx.def_path_str(key) }
+        }
+
+        /// Maps from the `DefId` of a trait to the list of
+        /// all the ancestors super traits.
+        query super_traits_of(key: DefId) -> FxHashSet<DefId> {
+            desc { |tcx| "computing the super traits of `{}`", tcx.def_path_str(key) }
+        }
+
+        /// Maps from the `DefId` of a trait to the list of
+        /// super-predicates. This is a subset of the full list of
+        /// predicates. We store these in a separate map because we must
```

Comment:
The part of this comment that begins with "We store these in a separate map..." kind of makes no sense and seems outdated. I think maybe it was copy-and-pasted from somewhere else?

Perhaps say:

The `Option<Ident>` is the name of an associated type. If it is `None`, then this query returns the full set of predicates. If `Some`, then the query returns only the subset of super-predicates that reference traits that define the given associated type. This is used to avoid cycles in resolving types like `T::Item`.